server/src/main/java/org/opensearch/gateway/remote/ClusterStateChecksum.java

```diff
@@ -0,0 +1,419 @@
+            nodesChecksum = checksumOut.getChecksum();
+
+            checksumOut.reset();
+            clusterState.coordinationMetadata().writeTo(checksumOut);
+            coordinationMetadataChecksum = checksumOut.getChecksum();
+
+            checksumOut.reset();
+            Settings.writeSettingsToStream(clusterState.metadata().persistentSettings(), checksumOut);
+            settingMetadataChecksum = checksumOut.getChecksum();
+
+            checksumOut.reset();
+            Settings.writeSettingsToStream(clusterState.metadata().transientSettings(), checksumOut);
+            transientSettingsMetadataChecksum = checksumOut.getChecksum();
+
+            checksumOut.reset();
+            clusterState.metadata().templatesMetadata().writeTo(checksumOut);
+            templatesMetadataChecksum = checksumOut.getChecksum();
+
+            checksumOut.reset();
+            clusterState.metadata().customs().forEach((key, custom) -> {
+                try {
+                    custom.writeTo(checksumOut);
+                } catch (IOException e) {
+                    logger.error("Failed to create checksum for custom metadata.", e);
+                    throw new RemoteStateTransferException("Failed to create checksum for custom metadata.", e);
+                }
+            });
+            customMetadataMapChecksum = checksumOut.getChecksum();
+
+            checksumOut.reset();
+            ((DiffableStringMap) clusterState.metadata().hashesOfConsistentSettings()).writeTo(checksumOut);
+            hashesOfConsistentSettingsChecksum = checksumOut.getChecksum();
+
+            checksumOut.reset();
+            clusterState.metadata().indices().forEach(((s, indexMetadata) -> {
+                try {
+                    indexMetadata.writeTo(checksumOut);
+                } catch (IOException e) {
+                    logger.error("Failed to create checksum for index metadata.", e);
+                    throw new RemoteStateTransferException("Failed to create checksum for index metadata.", e);
+                }
+            }));
+            indicesChecksum = checksumOut.getChecksum();
+
+            checksumOut.reset();
+            clusterState.blocks().writeTo(checksumOut);
+            blocksChecksum = checksumOut.getChecksum();
+
+            checksumOut.reset();
+            clusterState.customs().forEach((key, value) -> {
+                try {
+                    checksumOut.writeNamedWriteable(value);
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            });
+            clusterStateCustomsChecksum = checksumOut.getChecksum();
+        } catch (IOException e) {
+            logger.error("Failed to create checksum for cluster state.", e);
+            throw new RemoteStateTransferException("Failed to create checksum for cluster state.", e);
+        }
+        clusterStateChecksum = JZlib.crc32_combine(routingTableChecksum, nodesChecksum, 8);
```

Comment:
nit: the clusterStateCheckSum calculation could be moved to a separate function for abstraction.

Reply:
Also `8` could be made an constant.